src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((grouped, item) => {
+    const keyValue = item[key] as string;
+
+    if (!grouped[keyValue]) {
+      grouped[keyValue] = [];
+    }
+
+    grouped[keyValue].push(item);
+
+    return { ...grouped };
```

Comment:
```suggestion
    return grouped;
```
redundant to destructure grouped.